backend/module/eCampCore/src/EntityService/CampCollaborationService.php

```diff
@@ -45,37 +51,55 @@ protected function createEntity($data) {
 
         $authUser = $this->getAuthUser();
         if (!isset($data->userId)) {
+            $data->userId = null;
+        }
+        $inviteEmail = isset($data->inviteEmail) ? $data->inviteEmail : null;
+        if (null == $data->userId && !$inviteEmail) {
             $data->userId = $authUser->getId();
         }
 
         /** @var Camp $camp */
         $camp = $this->findRelatedEntity(Camp::class, $data, 'campId');
 
         /** @var User $user */
-        $user = $this->findRelatedEntity(User::class, $data, 'userId');
-
-        if (!isset($data->role)) {
-            $data->role = CampCollaboration::ROLE_MEMBER;
-        }
+        $user = null != $data->userId ? $this->findRelatedEntity(User::class, $data, 'userId') : null;
 
         $q = $this->fetchAllQueryBuilder();
-        $q->andWhere('row.camp = :campId');
-        $q->andWhere('row.user = :userId');
-        $q->andWhere('row.status = :status');
+        $expr = new Expr();
+        $q->andWhere($expr->andX(
+            $q->expr()->eq('row.camp', ':campId'),
+            $expr->orX(
+                $expr->eq('row.user', ':userId'),
+                $expr->eq('row.inviteEmail', ':inviteEmail')
+            )
+        ));
         $q->setParameter('campId', $camp->getId());
-        $q->setParameter('userId', $user->getId());
-        $q->setParameter('status', CampCollaboration::STATUS_LEFT);
+        $q->setParameter('userId', null != $user ? $user->getId() : null);
```

Comment:
If you save `null != $user ? $user->getId() : null` into a well-named variable, you can avoid calculating it again later and also make the code easier to read.

Reply:
calculating it again is not an issue. Performance optimization should only be done if really really necessary.
Better keep the code readable than performant.
I extracted the check into a variable.
As already mentioned, i would like to refactor later.